src/cpg_flow/inputs.py

```diff
@@ -0,0 +1,268 @@
+"""
+Metamist wrapper to get input sequencing groups.
+"""
+
+import logging
+
+from cpg_utils.config import config_retrieve, update_dict
+
+from cpg_flow.filetypes import CramPath, GvcfPath
+from cpg_flow.targets import Dataset, MultiCohort, PedigreeInfo, SequencingGroup, Sex
+
+from .metamist import (
+    AnalysisType,
+    Assay,
+    MetamistError,
+    get_cohort_sgs,
+    get_metamist,
+    parse_reads,
+)
+from .utils import exists
+
+_multicohort: MultiCohort | None = None
+
+
+def add_sg_to_dataset(dataset: Dataset, sg_data: dict) -> SequencingGroup:
+    """
+    This is moved here just to reduce code duplication
+
+    Args:
+        dataset (Dataset): Dataset to insert the SequencingGroup into
+        sg_data (dict): data from the metamist API
+
+    Returns:
+        The SequencingGroup object
+    """
+    # scavenge all the metadata from the SG dict (SG/Sample/Participant)
+    metadata = sg_data.get("meta", {})
+    update_dict(metadata, sg_data["sample"]["participant"].get("meta", {}))
+    # phenotypes are managed badly here, need a cleaner way to get them into the SG
+    update_dict(
+        metadata, {"phenotypes": sg_data["sample"]["participant"].get("phenotypes", {})}
+    )
+
+    # create a SequencingGroup object from its component parts
+    sequencing_group = dataset.add_sequencing_group(
+        id=str(sg_data["id"]),
+        external_id=str(sg_data["sample"]["externalId"]),
+        participant_id=sg_data["sample"]["participant"].get("externalId"),
+        meta=metadata,
+        sequencing_type=sg_data["type"],
+        sequencing_technology=sg_data["technology"],
+        sequencing_platform=sg_data["platform"],
+    )
+
+    if reported_sex := sg_data["sample"]["participant"].get("reportedSex"):
+        sequencing_group.pedigree.sex = Sex.parse(reported_sex)
+
+    # parse the assays and related dict content
+    _populate_alignment_inputs(sequencing_group, sg_data)
+
+    return sequencing_group
+
+
+def actual_get_multicohort() -> MultiCohort:
+    """Return the multicohort object"""
+    global _multicohort
+    if not _multicohort:
+        _multicohort = create_multicohort()
+    return _multicohort
+
+
+def get_multicohort() -> MultiCohort:
+    """
+    Return the cohort or multicohort object based on the workflow configuration.
+    """
+    input_datasets = config_retrieve(["workflow", "input_datasets"], [])
+    custom_cohort_ids = config_retrieve(["workflow", "input_cohorts"], [])
+    if custom_cohort_ids and input_datasets:
+        raise ValueError(
+            "Cannot use both custom_cohort_ids and input_datasets in the same workflow"
+        )
+
+    # NOTE: When configuring sgs in the config is deprecated, this will be removed.
+    if custom_cohort_ids:
+        if not isinstance(custom_cohort_ids, list):
+            raise ValueError("input_cohorts must be a list")
+        return actual_get_multicohort()
+    return get_multicohort()
```

Comment:
We can refactor this and rename actual_get_multicohort(). This set of code used to handle two cases, the actual_get_multicohort() method and a deprecated method. 

This looks like it will result in an infinite loop on line 88, if input cohorts are not specified. 

We should refactor this whole block and can likely simplify it significantly.